chainstate/tx-verifier/src/transaction_verifier/reward_distribution.rs

```diff
@@ -120,6 +136,51 @@ fn calculate_staker_reward(
     Some(staker_reward)
 }
 
+fn calculate_staker_reward_v1(
+    total_reward: Amount,
+    pool_balance: Amount,
+    staker_balance: Amount,
+    cost_per_block: Amount,
+    mpt: PerThousand,
+    pool_id: PoolId,
+) -> Result<Amount, ConnectTransactionError> {
+    debug_assert!(staker_balance <= pool_balance);
+
+    let staker_reward = match total_reward - cost_per_block {
+        Some(v) => {
```

Comment:
Not sure it's much of an improvement but I think using `let v = (total_reward - cost_per_block).unwrap_or(0);` will work without having to go into two separate code branches.

Note that there is currently a subtle difference in how the two code paths behave in case `pool_balance` is zero. An error is returned if and only if `total_reward >= cost_per_block` which seems like an irrelevant condition. Unifying the two paths will make the behaviour more consistent.

Also, have you thought about a more descriptive name for `v`?